### DISCUSSION TITLE
Request: Proxy Support for Web Search on `pip`

### DISCUSSION THREAD

**Discussion by AeneasZhu** [open-webui/open-webui#4346] [General]
Open Webui on `pip` doesn't support proxy when using web search. Unlike on `docker`, where proxy can be set in docker config panel, there is no stull like `proxy setting` on the setting panel. Since most foreign websites like google, bbc are blocked in the country, web searching via proxy is the only choice. Will open-webui add support for proxy like socks5 setting in the future?

**Comment by Mushy-Snugglebites-badonkadonk** [comment]
Is this what you’re looking for?

https://docs.openwebui.com/getting-started/env-configuration#proxy-settings

Editing the Docker compose file to input the variable `http_proxy=http://[IP]:[PORT]`, OWUI boots up, but downloaded models will not show up in the drop down. Only OpenAI APIs that are set up in _Settings_. Setting the `http_proxy` breaks the connection to the Ollama docker instance. As a result, I am stuck.

**Goal**: To proxy website retrieval when using _web search_.

**Reply by ips972** [reply]
you have to add more ENV params like no_http stating your ollama and local net- so http_proxy bypass those destinations.
